File: src/modules/AvatarView/ios/AvatarViewManager.m
#import <React/RCTViewManager.h>

/**
 * Objective-C bridge that exposes the Swift AvatarViewManager to the
 * React Native bridge and declares the `name` prop.
 *
 * The macro RCT_EXTERN_MODULE links to the Swift class.
 * RCT_EXTERN__REMAP_METHOD is used for custom prop setters.
 *
 * Place this file in the same iOS target as AvatarViewManager.swift and
 * AvatarUIView.swift.
 */
RCT_EXTERN_MODULE(AvatarViewManager, RCTViewManager)

RCT_EXTERN_METHOD(name : (NSString *)name)

File: src/modules/AvatarView/ios/AvatarViewManager.swift
import Foundation
import React

/**
 * RCTViewManager that exposes AvatarUIView to the React Native bridge.
 *
 * Registration:
 *   Swift modules are auto-registered when they conform to RCTBridgeModule.
 *   No additional Obj-C boilerplate needed beyond AvatarViewManager.m.
 *
 * Usage (via AvatarView.tsx wrapper):
 *   <AvatarView name="Santiago Lopez" size={44} />
 */
@objc(AvatarViewManager)
final class AvatarViewManager: RCTViewManager {

    override func view() -> UIView! {
        return AvatarUIView()
    }

    /// Tells React Native this view manager should be used on the main queue.
    override static func requiresMainQueueSetup() -> Bool {
        return true
    }
}
